processlib/activity.py

```diff
@@ -327,34 +347,31 @@ class StartViewActivity(StartMixin, ViewActivity):
 
 class EndActivity(Activity):
     def instantiate(self, **kwargs):
-        super(EndActivity, self).instantiate(**kwargs)
+        super().instantiate(**kwargs)
         self.start()
-        self.finish()
+        # Finish activity
+        self.finish(**kwargs)
+        # Finish complete process
+        self.finish_process()
 
-    def finish(self, **kwargs):
-        super(EndActivity, self).finish(**kwargs)
 
-        update_fields = []
-        if not self.process.finished_at:
-            self.process.finished_at = self.instance.finished_at
-            update_fields.append("finished_at")
-
-        if not self.process.status == self.process.STATUS_DONE:
-            self.process.status = self.process.STATUS_DONE
-            update_fields.append("status")
-
-        self.process.save(update_fields=update_fields)
+class EndRedirectActivity(Activity):
```

Comment:
Kann das jetzt nicht einfach genau die gleiche sein wie die `EndActivity` mit dem einzigen Unterschied, dass sie ein redirect_url_callback verwendet. Eigentlich aber könnte man auch die reguläre EndActivity jetzt verwenden oder?

Also braucht es die ganze custom implementierung unten mit eigener finish etc überhaupt noch?

Reply:
An sich kann dies auch in der EndActivity behandelt werden. Ich wollte es für die Abwärtskompatiblität jetzt nicht komplett umbauen

Reply:
Aber reicht es da nicht, dass die EndRedirectActivity einfach die EndActivity erbt, das rückwärtskompatible callback erlaubt und ansonsten nichts tut?

Reply:
Stimmt, passt es für dich mit complete_redirect_url, oder wäre dir der redirect_url_callback lieber? Ich finde dass es mit der neuen Variante besser mit der Message zusammen passt

Reply:
passt für mich :-)